### PR TITLE
feat(create-axiom): engine-native starter and demo onboarding redesign

### PR DESCRIPTION
**Pull request by naml14**
## Summary

Engine-native redesign of the `create-axiom` starter and the demo onboarding flow: new starter templates (app, styles, dev-server), externalized demo launcher, and aligned SSR render. Includes the animation composed-transform fallback fix and reflow coverage.

> **Chained PR 1 of 3 — base `main`.** Stacked-to-main: each slice lands independently in order. Merge this first.

### Dependency chain
```
main
 └─ 📍 PR #A  chain/onboarding-starter      (this PR)
     └─ PR #B  chain/security-c2-batch1
         └─ PR #C  chain/hardening-c2-batch2
```

### Review budget
- **Code (`src/` + `scripts/`): 336 lines** · tests: 332 · OpenSpec design docs: 245 · total additions+deletions ≈ 913.
- `size:exception` rationale: the over-budget portion is the starter **template files + their tests** (cohesive, low-risk) plus co-located OpenSpec design rationale. Actual logic to scrutinize is ~336 lines. Focus review on `scripts/templates/*`, `scripts/create-axiom.ts`, `src/features/animation.ts`.

## Type of change
- [x] `feat` — new feature (minor bump)

## Checklist
- [x] Tests added or updated for the changed behavior
- [x] `bun test` passes locally (full suite green on the integrated chain)
- [x] `bun run typecheck` passes locally
- [x] This PR links an issue (`Part of #30`)
- [ ] `docs/V1-0-0-EVIDENCE-LOG.md` — N/A (absent in branch)
- [x] Conventional Commits
- [x] Hot path invariant respected (no DOM reads added to reflow/fast-path/flex/commit)
- [x] No runtime dependencies added to `src/`

## Related issues
Part of #30

## Testing notes
Final integrated tree of the chain is byte-identical to `feat/starter-demo-onboarding` (verified: empty `git diff`). Per-slice CI validates each PR independently.

## Summary by Sourcery

Rediseñar el starter `create-axiom` y la demo de onboarding para usar helpers nativos del engine, reforzar los contratos de SSR/animación y robustecer la experiencia de desarrollo/demo.

New Features:
- Introducir una nueva plantilla de aplicación starter nativa del engine con secciones interactivas de contador, lista y siguientes pasos conectadas a los helpers de layout.
- Agregar una plantilla de servidor de desarrollo basado en Bun con script externo de lanzamiento compatible con CSP y cabeceras de seguridad, además de soporte de instalación de dependencias en el scaffolder.

Bug Fixes:
- Corregir el comportamiento de transformaciones compuestas añadiendo un valor de reserva seguro `var(--animation-transform,)` para que el layout siga siendo válido cuando la variable CSS no está definida.
- Asegurar que los nodos de texto raíz se midan correctamente en el reflow, incluyendo las rutas `textEngine` y de reserva, preservando al mismo tiempo los nodos de texto vacíos de altura cero.

Enhancements:
- Actualizar la demo de portal para usar APIs de helpers idiomáticas (`h`, `stack`, `row`) manteniendo el patrón del root de portal dedicado.
- Mejorar la hoja de estilos del starter con nuevos tokens y clases utilitarias alineadas con la UI actualizada del starter.
- Validar los nombres de proyecto en `create-axiom` y personalizar el HTML generado y los encabezados con el nombre de proyecto elegido.
- Alinear la salida de la cadena de transformación en SSR con el renderizador del cliente para preservar el contrato de transformaciones compuestas durante la hidratación.
- Documentar los nuevos puntos de entrada del starter y de la demo de portal en el README y regenerar la caché de metadatos del registro de skills.

Tests:
- Ampliar las pruebas de `create-axiom` para cubrir la existencia de archivos de plantilla, la personalización, la instalación de dependencias y las cabeceras de seguridad del servidor de desarrollo.
- Agregar cobertura para la estructura y convenciones de estilo de la demo de portal, la guía de onboarding del README, los casos límite de reflow y el contrato actualizado de cadenas de transform/SSR.

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Redesign the create-axiom starter and demo onboarding to use engine-native helpers, strengthen SSR/animation contracts, and harden the dev/demo experience.

New Features:
- Introduce a new engine-native starter app template with interactive counter, list, and next-steps sections wired to layout helpers.
- Add a Bun-powered dev server template with CSP-friendly external launcher script and security headers, plus dependency installation support in the scaffolder.

Bug Fixes:
- Fix composed transform behavior by adding a safe var(--animation-transform,) fallback so layout remains valid when the CSS variable is unset.
- Ensure root text nodes are correctly measured in reflow, including textEngine and fallback paths, while preserving zero-height empty text nodes.

Enhancements:
- Refresh the portal demo to use idiomatic helper APIs (h, stack, row) while preserving the dedicated portal root pattern.
- Improve the starter stylesheet with new tokens and utility classes aligned to the updated starter UI.
- Validate project names in create-axiom and personalize generated HTML and headings with the chosen project name.
- Align SSR transform string output with the client renderer to preserve the composed transform contract across hydration.
- Document the new starter and portal demo entry points in the README and regenerate the skill registry metadata cache.

Tests:
- Expand create-axiom tests to cover template file presence, personalization, dependency installation, and dev server security headers.
- Add coverage for portal demo structure and style conventions, README onboarding guidance, reflow edge cases, and the updated transform/SSR string contract.

</details>

### CODE REVIEW

File: tests/create-axiom.test.ts [hunk 1]
```diff
@@ -134,15 +170,92 @@
...
+
+		// Title is personalized with the project name
+		expect(indexHtml).toContain("<title>my-app</title>");
+		expect(indexHtml).not.toContain("{{PROJECT_NAME}}");
+
+		// No raw className — always class
+		expect(starterApp).not.toContain("className");
+
+		// New starter markers
+		expect(starterApp).toContain("Edit src/app.ts to start building")
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/create-axiom.test.ts]: Corregido en 300163a: agregue el punto y coma faltante en la linea del expect(...). Gracias por el catch.

File: tests/portal-demo.test.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { describe, expect, test } from "bun:test";
+import { readFile } from "node:fs/promises";
+import { join, dirname } from "node:path";
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/portal-demo.test.ts]: Corregido en 802e72c: elimine el import dirname que no se usaba. (Mismo hallazgo que reporto Copilot, ambos atendidos.)

File: scripts/templates/dev-server.ts [hunk 3]
```diff
@@ -90,29 +90,82 @@ async function resolvePublicAsset(pathname: string): Promise<string | null> {
...
 	return new Response(file, {
 		headers: {
 			"Content-Type": file.type || "application/octet-stream",
+			...SECURITY_HEADERS,
+			...corsHeaders(req),
 		},
 	});
 }
 
+const ALLOWED_ORIGINS = ["http://localhost:5173"];
```
[the remark this answers is not in the document]
Comment (naml14) [reply scripts/templates/dev-server.ts]: Aplicado en 78ef0b2: agregue http://127.0.0.1:5173 a ALLOWED_ORIGINS junto a localhost, ya que Vite responde en ambos hosts.

File: scripts/templates/dev-server.ts [hunk 4]
```diff
@@ -90,29 +90,82 @@ async function resolvePublicAsset(pathname: string): Promise<string | null> {
...
+		new URL(origin);
+	} catch {
+		return {};
+	}
+
+	if (origin === "*") return {};
+
+	if (!ALLOWED_ORIGINS.includes(origin)) {
+		return {};
+	}
```
[the remark this answers is not in the document]
Comment (naml14) [reply scripts/templates/dev-server.ts]: Confirmado y corregido en 78ef0b2: era dead code. new URL('*') lanza y el catch ya retorna {}, asi que el check origin === '*' nunca se alcanzaba. Eliminado.

File: scripts/create-axiom.ts [hunk 5]
```diff
@@ -59,6 +61,14 @@ export async function scaffoldProject(
 			final = `${JSON.stringify(pkg, null, 2)}\n`;
 		}
 
+		if (src === "index.html") {
+			final = content.replace("{{PROJECT_NAME}}", projectName);
+		}
+
+		if (src === "src/app.ts") {
+			final = content.replace("'{{PROJECT_NAME}}'", `'${projectName}'`);
+		}
```
[the remark this answers is not in the document]
Comment (naml14) [reply scripts/create-axiom.ts:79]: Aplicado en 300163a, con un matiz honesto: hoy cada template tiene una sola ocurrencia de {{PROJECT_NAME}}, asi que no era un bug actual. Igual cambie a replaceAll como hardening por si se agregan mas placeholders en el futuro.

File: tests/create-axiom.test.ts [hunk 6]
```diff
@@ -134,15 +170,92 @@ describe("create-axiom starter", () => {
...
+	test("scaffoldProject personalizes the HTML title with the given project name", async () => {
+		const workspace = await freshDir("title");
+		const projectDir = join(workspace, "cool-project");
+		await scaffoldProject(projectDir, "cool-project");
+
+		const indexHtml = await readFile(join(projectDir, "index.html"), "utf8");
+
+		expect(indexHtml).toContain("<title>cool-project</title>");
+		expect(indexHtml).not.toContain("{{PROJECT_NAME}}");
+		expect(indexHtml).not.toContain("my-app");
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/create-axiom.test.ts:261]: Hecho en 300163a: extraje isValidProjectName() como funcion pura y agregue cobertura de nombres validos e invalidos (incluyendo intentos de path traversal e inyeccion de shell).

File: tests/create-axiom.test.ts [hunk 7]
```diff
@@ -119,10 +119,46 @@ afterEach(async () => {
...
+	test("installProjectDependencies returns subprocess exit code", async () => {
+		const projectDir = await freshDir("install-exit");
+		await writeFile(
+			join(projectDir, "package.json"),
+			`${JSON.stringify({ name: "tmp-install", private: true }, null, 2)}\n`,
+			"utf8",
+		);
+
+		const exitCode = installProjectDependencies(projectDir);
+		expect(typeof exitCode).toBe("number");
```
[the remark this answers is not in the document]
Comment (naml14) [reply tests/create-axiom.test.ts]: Hecho en 300163a: el test ahora usa un spy sobre Bun.spawnSync (sin bun install real), verifica los args ['bun','install'] y el cwd, y agregue cobertura del path de exit code distinto de cero.

File: scripts/templates/dev-server.ts [hunk 8]
```diff
@@ -90,29 +90,82 @@ async function resolvePublicAsset(pathname: string): Promise<string | null> {
...
-function serveStaticFile(filePath: string): Response {
+const SECURITY_HEADERS: Record<string, string> = {
+	"X-Content-Type-Options": "nosniff",
+	"X-Frame-Options": "SAMEORIGIN",
+	"Referrer-Policy": "strict-origin-when-cross-origin",
+	"Content-Security-Policy": "default-src 'self'; script-src 'self'",
+	"Permissions-Policy": "geolocation=(), camera=(), microphone=()",
+};
+
+function serveStaticFile(filePath: string, req: Request): Response {
```
[the remark this answers is not in the document]
Comment (naml14) [reply scripts/templates/dev-server.ts:101]: Buena observacion. La dejo fuera de esta tanda de fixes de review a proposito: un helper de respuesta compartido tocaria todas las rutas del dev-server y merece su propio PR enfocado para no mezclar un refactor estructural con estos fixes puntuales. La dejo abierta para que lo decidas.

File: tests/portal-demo.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (naml14) [reply tests/portal-demo.test.ts]: Corregido en 802e72c: elimine el import dirname. (Mismo hallazgo que reporto CodeQL, ambos atendidos.)
